dingo/MetabolicNetwork.py

```diff
@@ -223,3 +223,205 @@ def shut_down_reaction(self, index_val):
 
         self._lb[index_val] = 0
         self._ub[index_val] = 0
+
+
+class CommunityMetabolicNetwork():
+
+    # This implementation works only for communities of 2 models 
+    # Once our method is validated, we will move on to implement it for more 
+
+    def __init__(self, comm_tuple_args):
+
+        self._comm_parameters = {}
+        self._comm_parameters["opt_percentage"]   = 100
+        self._comm_parameters["distribution"]     = "uniform"
+        self._comm_parameters["nullspace_method"] = "sparseQR"
+
+        try:
+            import gurobipy
+
+            self._comm_parameters["fast_computations"] = True
+        except ImportError as e:
+            self._comm_parameters["fast_computations"] = False
+
+        if len(comm_tuple_args) != 8:
+            raise Exception(
+                "An unknown input format given to initialize a metabolic network object."
+            )
+
+        self._comm_lb               = comm_tuple_args[0]
+        self._comm_ub               = comm_tuple_args[1]
+        self._comm_S                = comm_tuple_args[2]
+        self._comm_metabolites      = comm_tuple_args[3]
+        self._comm_reactions        = comm_tuple_args[4]
+        self._comm_biomass_index    = comm_tuple_args[5]
+        self._comm_biomass_function = comm_tuple_args[6]
+        self._modelList        = comm_tuple_args[7]
+
+        try:
+            if (
+                self._comm_lb.size                  != self._comm_ub.size
+                or self._comm_lb.size               != self._comm_S.shape[1]
+                or len(self._comm_metabolites)      != self._comm_S.shape[0]
+                or len(self._comm_reactions)        != self._comm_S.shape[1]
+                or self._comm_biomass_function.size != self._comm_S.shape[1]
+                or (self._comm_biomass_index < 0)
+                or (self._comm_biomass_index > self._comm_biomass_function.size)
+            ):
+                raise Exception(
+                    "Wrong tuple format given to initialize a metabolic network object."
+                )
+        except LookupError as error:
+            raise error.with_traceback(sys.exc_info()[2])
+
+
+    @classmethod
+    def buildModelList(cls, directory, format_type):
+        comm_tuple_args = getModelList(directory, format_type)
+        return cls(comm_tuple_args)
+
+    def fva(self):
+        """A member function to apply the FVA method on the community metabolic network."""
+
+        if self._comm_parameters["fast_computations"]:
```

Comment:
You should set the objective function to the zero vector before FVA() to study the unbiased case.